shapedetector.py

```diff
@@ -77,15 +80,24 @@ def detect_shapes():
             c = c.astype(int)
             cv2.drawContours(image, [c], -1, (255, 0, 0), 2)
             cv2.putText(image, shape, (cx, cy), cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 0, 0), 2)
-
-            # cv2.imshow("Image", image)
-            # cv2.waitKey(0)
-
+	
```

Comment:
you should consider putting this part of the script that is after all the function generators inside 
```python
if __name__ == "__main__":
```

this avoids your script being run if someone were to import this file.  it should probably only be run if it is the fun run

Reply:
i'm a little confused why you put the if clause around just a few lines.  assuming im interpreting your code correctly, the `if __name__` ... should be on this line